build/Build.cs

```diff
@@ -0,0 +1,137 @@
+using Nuke.Common;
+using Nuke.Common.CI;
+using Nuke.Common.Execution;
+using Nuke.Common.Git;
+using Nuke.Common.IO;
+using Nuke.Common.ProjectModel;
+using Nuke.Common.Tooling;
+using Nuke.Common.Tools.DotNet;
+using Nuke.Common.Tools.NuGet;
+using Nuke.Common.Utilities.Collections;
+using Nuke.OctoVersion;
+using OctoVersion.Core;
+using static Nuke.Common.EnvironmentInfo;
+using static Nuke.Common.IO.FileSystemTasks;
+using static Nuke.Common.IO.PathConstruction;
+using static Nuke.Common.Tools.DotNet.DotNetTasks;
+
+[CheckBuildProjectConfigurations]
+[ShutdownDotNetAfterServerBuild]
+class Build : NukeBuild
+{
+    [Parameter("Configuration to build - Default is 'Debug' (local) or 'Release' (server)")]
+    readonly Configuration Configuration = IsLocalBuild ? Configuration.Debug : Configuration.Release;
+
+    [NukeOctoVersion] readonly OctoVersionInfo OctoVersionInfo;
+
+    [Solution(GenerateProjects = true)] readonly Solution Solution;
+
+    [Parameter("Test filter expression", Name = "where")] readonly string TestFilter = string.Empty;
+    [Parameter, Secret] readonly string FeedzIoApiKey;
+    [Parameter, Secret] readonly string NuGetApiKey;
+
+    AbsolutePath SourceDirectory => RootDirectory / "source";
+    AbsolutePath ArtifactsDirectory => RootDirectory / "artifacts";
+    AbsolutePath LocalPackagesDirectory => RootDirectory / ".." / "LocalPackages";
+
+    Target Clean => _ => _
+        .Before(Restore)
+        .Executes(() =>
+        {
+            SourceDirectory.GlobDirectories("**/bin", "**/obj", "./source/**/TestResults").ForEach(DeleteDirectory);
+            EnsureCleanDirectory(ArtifactsDirectory);
+        });
+
+    Target Restore => _ => _
+        .DependsOn(Clean)
+        .Executes(() =>
+        {
+            DotNetRestore(s => s
+                .SetProjectFile(Solution));
+        });
+
+    Target Compile => _ => _
+        .DependsOn(Restore)
+        .Executes(() =>
+        {
+            DotNetBuild(s => s
+                .SetProjectFile(Solution)
+                .SetConfiguration(Configuration)
+                .SetVersion(OctoVersionInfo.FullSemVer)
+                .SetAssemblyVersion(OctoVersionInfo.MajorMinorPatch)
+                .SetInformationalVersion(OctoVersionInfo.InformationalVersion)
+                .EnableNoRestore());
+        });
+
+    Target Test => _ => _
+        .DependsOn(Compile)
+        .Executes(() =>
+        {
+            DotNetTest(_ => _
+                .SetProjectFile(Solution)
+                .SetConfiguration(Configuration)
+                .SetLoggers("trx")
+                .SetVerbosity(DotNetVerbosity.Normal)
+                .SetFilter(TestFilter)
+                .EnableNoBuild()
+                .EnableNoRestore());
+            GlobFiles(SourceDirectory, "**/*.trx")
+                .ForEach(x => CopyFileToDirectory(x, ArtifactsDirectory, FileExistsPolicy.Overwrite));
+        });
+
+    Target Pack => _ => _
+        .DependsOn(Compile)
```

Comment:
Not required, test depends on Compile already

Reply:
I wanted to be explicit about dependencies here, rather than relying on knowledge about how the dependency is configured.
Feels like a better pattern - we can avoid each target knowing too much

Reply:
> Feels like a better pattern - we can avoid each target knowing too much

I'm not quite sure I understand what you mean by this. In this scenario, your target knows _more_ than I needs to since it has a dependency on Compile too.

Reply:
`Pack` does know that it depends on compile, because it needs the output of the binaries.
It also knows that it needs tests to succeed.
it _doesn't_ need to know that `Test` requires the binaries.

_am i making sense?_